TYPO3.Neos/Resources/Private/Templates/Module/Management/Workspaces/Show.html

```diff
@@ -108,11 +108,18 @@
 						$('tbody input[type="checkbox"]').prop('checked', value);
 					});
 					$('tbody input[type="checkbox"]').change(function() {
-						if ($(this).closest('tr').attr('data-ismoved') == 'true') {
+						if ($(this).closest('tr').data('ismoved') === true) {
 							var currentNodePath = $(this).closest('tr').attr('data-nodepath') + '/';
 							var checked = $(this).prop('checked');
-							$('tr[data-nodepath^="' + currentNodePath + '"] td.check input').each(function(index, element) {
-								$(element).prop('checked', checked);
+
+							function nodePathStartsWith(nodePath) {
+								return function(index, element) {
+									return nodePath.indexOf($(element).data('nodepath')) === 0;
+								}
+							}
+							var movedParentDocuments = $(this).closest('table').find('.neos-document[data-ismoved="true"]').filter(nodePathStartsWith(currentNodePath));
+							$(movedParentDocuments).each(function(index, movedParentDocument) {
+								$('tr[data-nodepath^="' + $(movedParentDocument).data('nodepath') + '"] td.check input').prop('checked', checked);
```

Comment:
Thanks, but that doesn't work. What I need to do here is finding the moved document upwards the tree, not downwards.

	TASK: Adjust user interface to auto-select moved parent documents  …
This adds a behaviour to the selection in the review view, which
automatically selects and deselects all nodes of the branch which
whose top document node has been moved.

> Example: Consider a document node structure of A1 / B2 / B3 / B4. The
> node B has been moved. If B3 or B4 are selected, the whole B-branch
> will be selected, because B3 or B4 can't be published or discarded own
> their own - without B2 being published or discarded as well, B3 or B4
> would become orphaned nodes.

If it's not bad JS style, I'd like to leave it like that, because as far as I can see, it works as intended.